torch/csrc/jit/frontend/schema_matching.cpp

```diff
@@ -20,6 +20,15 @@ static inline bool isIntOrFloatUsedAsList(
     const Argument& arg) {
   // Look for int[N] or float[N]
   const auto& v_type = value->type();
+  // Check if value is optional Type of Int or Float
+  if (value->type()->kind() == OptionalType::Kind) {
```

Comment:
Most of logic here is identical to non-optional case. Could we simplify the implementation a little bit to something like this?
```
if (value->type()->kind() == OptionalType::Kind) {
    v_type = unwrapOptional(v_type);  // You may need to remove `const` modifier of `v_type`.
}

... original logic stay as is ...
```

Reply:
Agreed and fixed.